### ISSUE TITLE
Progress bar fails when only stderr is not a TTY

### ISSUE THREAD

**Issue by fson** [yarnpkg/yarn#2200]
**Do you want to request a *feature* or report a *bug*?**

Bug

**What is the current behavior?**

Before trying to write a progress bar, Yarn checks if **stdout** is a TTY, but writes the progress to **stderr**. When `yarn add` is spawned as a child process with a non-TTY stderr (e.g. using the `pipe` or `ignore` option for `stderr`), it fails unless also stdout is not a TTY.

Following script fails:
```js
const { spawn } = require('child_process');

const yarn = spawn('yarn', ['add', 'react'], {
  stdio: [process.stdin, process.stdout, 'pipe']
})
yarn.stderr.on('data', (data) => {
  console.log(data.toString());
})
```
yarn-error.log:
```
Arguments:
  /Users/ville/.nvm/versions/node/v6.9.1/bin/node /Users/ville/.nvm/versions/node/v6.9.1/bin/yarn add react

PATH:
  /Users/ville/.rvm/gems/ruby-2.3.1/bin:/Users/ville/.rvm/gems/ruby-2.3.1@global/bin:/Users/ville/.rvm/rubies/ruby-2.3.1/bin:/usr/local/heroku/bin:/Users/ville/.nvm/versions/node/v6.9.1/bin:/Users/ville/.opam/4.02.3/bin:/Library/Frameworks/Python.framework/Versions/3.5/bin:/usr/local/bin:/usr/local/sbin:/Users/ville/bin:/Users/ville/android-sdk-macosx/tools:/Users/ville/android-sdk-macosx/platform-tools:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/ville/.rvm/bin

Yarn version:
  0.17.10

Node version:
  6.9.1

Platform:
  darwin x64

npm manifest:
  No manifest

yarn manifest:
  No manifest

Lockfile:
  No lockfile

Trace:
  TypeError: Expected `count` to be a positive finite number
      at module.exports (/Users/ville/.nvm/versions/node/v6.9.1/lib/node_modules/yarn/node_modules/repeating/index.js:10:9)
      at ProgressBar.render (/Users/ville/.nvm/versions/node/v6.9.1/lib/node_modules/yarn/lib/reporters/console/progress-bar.js:58:22)
      at ConsoleReporter.progress (/Users/ville/.nvm/versions/node/v6.9.1/lib/node_modules/yarn/lib/reporters/console/console-reporter.js:396:9)
      at /Users/ville/.nvm/versions/node/v6.9.1/lib/node_modules/yarn/lib/package-fetcher.js:123:36
      at next (native)
      at step (/Users/ville/.nvm/versions/node/v6.9.1/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:17:30)
      at /Users/ville/.nvm/versions/node/v6.9.1/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:35:14
      at Promise.F (/Users/ville/.nvm/versions/node/v6.9.1/lib/node_modules/yarn/node_modules/core-js/library/modules/_export.js:35:28)
      at /Users/ville/.nvm/versions/node/v6.9.1/lib/node_modules/yarn/node_modules/babel-runtime/helpers/asyncToGenerator.js:14:12
      at PackageFetcher.init (/Users/ville/.nvm/versions/node/v6.9.1/lib/node_modules/yarn/lib/package-fetcher.js:158:7)
```

**What is the expected behavior?**

Yarn should check that **stderr** is a TTY before writing the progress bar to it.

**Comment by fson** [comment]
I've made a PR with a fix for this issue: #2201.

**Comment by pascalberger** [comment]
This seems like a duplicate of #2010

**Comment by BYK** [comment]
Duplicate of #2010. Also, should be fixed by #4742.
